### PR TITLE
Update saml gems

### PR DESCRIPTION
**Pull request by pkarman**
**Why**:

* ruby-saml official release now has our changes
* saml_idp 18F fork is canonical

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,7 +25,7 @@ group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug'
   gem 'rspec-rails', '~> 3.4'
-  gem 'saml_idp', '~> 0.4.0'
+  gem 'saml_idp', git: 'https://github.com/18F/saml_idp.git', branch: 'master'
```
Comment (amoose) [inline Gemfile]: shorthand for github hosted: `github: '18F/saml_idp', branch: 'master'`

Comment (amoose) [inline Gemfile]: and, even though we're controlling that repo, shouldn't retain the pessimistic version operator?

Comment (pkarman) [inline Gemfile]: maybe we should start tagging our own "releases"? they won't be on rubygems but will give us some sense of checkpointing.

Comment (pkarman) [inline Gemfile]: codeclimate didn't like the `github` syntax which I think relies on ssh keys. The explicit https url allows anonymous checkout, afaik.

Comment (monfresh) [inline Gemfile]: It's actually a bundler-audit offense. We must explicitly use HTTPS URLs for securitay.

**Pull request merged**